payment/datastore.go

```diff
@@ -143,6 +146,64 @@ func (pg *Postgres) GetOrder(orderID uuid.UUID) (*Order, error) {
 	return &order, nil
 }
 
+// GetPagedMerchantTransactions - get a paginated list of transactions for a merchant
+func (pg *Postgres) GetPagedMerchantTransactions(
+	ctx context.Context, merchantID uuid.UUID, pagenation *inputs.Pagination) (*[]Transaction, int, error) {
+	var count int
+
+	countStatement := `
+			SELECT count(t.*) 
+			FROM transactions 
+				INNER JOIN order ON order.id = transaction.order_id
+			WHERE order.merchant_id = $1`
+
+	// get the total count
+	err := pg.DB.Select(&count, countStatement, merchantID)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	getStatement := `
+		SELECT t.*
+		FROM transactions
+			INNER JOIN order ON order.id = transaction.order_id
+		WHERE order.merchant_id = $1`
+
+	// $ numbered params for query
+	params := []interface{}{
+		merchantID,
+	}
+
+	start := 1
+	orderBy, v := pagenation.GetOrderBy(start)
+	count = len(v) + start
+	if orderBy != "" {
+		getStatement += orderBy
+		params = append(params, v...)
+	}
+
+	offset := pagenation.Page * pagenation.Items
+	if offset > 0 {
+		getStatement += fmt.Sprintf(" OFFSET $%d", count)
+		count++
```

Comment:
What's the incrementation for?

Reply:
We have no idea how the caller may wish to order by, so we need to dynamically structure the sql query to accommodate.  The count is the parameter count within the query `$1 $2` etc.